homeassistant/components/humidifier/__init__.py

```diff
@@ -170,13 +172,21 @@ def state_attributes(self) -> dict[str, Any]:
         if supported_features & HumidifierEntityFeature.MODES:
             data[ATTR_MODE] = self.mode
 
+        if self.current_humidity is not None:
+            data[ATTR_CURRENT_HUMIDITY] = self.current_humidity
```

Comment:
We don't want measurements to be state attributes. These should just be separate sensor entities with our current design. Some older integrations still have the legacy pattern of exposing them as state attributes.

Reply:
Thanks, Martin! I didn't realize the `ClimateEntity` was not a good example to reference given that it uses the legacy pattern. Would you suggest that `generic_hygrostat` expose a device instead of just an entity? The current humidity would then be exposed as a sensor within the device. This would allow integrations like HomeKit to pick up the current humidity correctly using [this](https://github.com/home-assistant/core/blob/e9697872c8ebb8e0b41c5e7bf78dab6d3b58afcf/homeassistant/components/homekit/__init__.py#L954-L961) code.

Reply:
Maybe. It would require a config flow to set up the integration. I think it's a question for our architecture discussion forum:

https://github.com/home-assistant/architecture/discussions

Please search before posting for related discussions.

Until the architecture discussion is resolved please close this PR. Thanks!

Reply:
Thanks! Started a discussion at https://github.com/home-assistant/architecture/discussions/783. Feel free to chime in with your thoughts. Closing this PR.